lib/cylc/network/httpclient.py

```diff
@@ -259,6 +259,7 @@ def put_messages(self, payload):
                 event_time (str): Event time as string.
                 messages (list): List in the form [[severity, message], ...].
         """
+        self._load_contact_info()
```

Comment:
The contact file is loaded at line 274 via `_compat` which means this change will load it twice.

I think we can avoid this something like:

```python
func_name = self._compat('put_messages')
```

Then later:

```python
if not func_name:
    func_name = self._compat('put_messages')
```

And later still:

```python
self.comms1 = {}
func_name = None
```

Reply:
Does https://github.com/cylc/cylc-flow/pull/4518/commits/69b20f155bf6804f4b710580069a33568c40ab3a answer?

Reply:
Not quite, that will always fail for retries.

Reply:
Oops - so https://github.com/cylc/cylc-flow/pull/4518/commits/afc67156ee280ccac95a8b5111c6ead68a4c6de5